shipment_advice_reception_planner/wizards/plan_reception_shipment.py

```diff
@@ -0,0 +1,106 @@
+    move_ids = fields.Many2many(
+        comodel_name="stock.move",
+        string="Moves Being Planned",
+    )
+    shipment_advice_id = fields.Many2one(
+        comodel_name="shipment.advice",
+        string="Shipment Advice",
+        required=True,
+    )
+    warehouse_id = fields.Many2one(related="shipment_advice_id.warehouse_id")
+    move_to_split_ids = fields.Many2many(comodel_name="wizard.stock.move.split")
+    partner_id = fields.Many2one(string="Receive From", comodel_name="res.partner")
+
+    @api.onchange("move_ids")
+    def _onchange_quantity_to_split_wizard(self):
+        self.move_to_split_ids = [(5, 0, 0)]
+        for move in self.move_ids:
+            if move.quantity_to_split_wizard != move.product_uom_qty:
+                self.move_to_split_ids = [
+                    (
+                        0,
+                        0,
+                        {
+                            "move_id": move.id,
+                            "quantity_to_split": move.quantity_to_split_wizard,
+                        },
+                    )
+                ]
+        return
+
+    @api.model
+    def default_get(self, fields_list):
+        res = super().default_get(fields_list)
+        active_ids = self.env.context.get("active_ids")
+        # Check shipment advice used for planning the reception
+        shipment_id = False
+        if active_ids:
+            shipment_id = active_ids[0]
+        shipment = self.env["shipment.advice"].browse(shipment_id)
+        if not shipment.exists():
+            raise UserError(_("No shipment advice found to plan the reception!"))
+        if shipment.shipment_type != "incoming":
+            raise UserError(
+                _("Only an incoming shipment can be used to plan a reception!")
+            )
+        if not shipment.arrival_date:
+            raise UserError(
+                _(
+                    "Please set an arrival date on the shipment before planning. "
+                    "It can still be changed later."
+                )
+            )
+        res["shipment_advice_id"] = shipment_id
+        partner_id = self.env.context.get("reception_planner_receive_from")
+        if partner_id:
+            partner = self.env["res.partner"].browse(partner_id)
+            if partner.exists():
+                res["partner_id"] = partner.id
+        return res
+
+    def _prepare_move_split_vals(self, move, split_qty):
+        split_qty_uom = move.product_id.uom_id._compute_quantity(
+            split_qty, move.product_uom, rounding_method="HALF-UP"
+        )
+        return move._prepare_move_split_vals(split_qty_uom)
+
+    def _split_move(self, move, split_qty):
+        split_move = move.copy(self._prepare_move_split_vals(move, split_qty))
+        new_product_qty = move.product_id.uom_id._compute_quantity(
+            move.product_qty - split_qty, move.product_uom, round=False
+        )
+        move.product_uom_qty = new_product_qty
+        return split_move
```

Comment:
This should be moved to `wizard.stock.move.split` then you can get rid of the arguments.

Reply:
Yes but it also works like that, no ?

Reply:
Yes of course it is working like that. 
It is just a matter of code style, it would simplify the code. 
You would than have the `_split_move` in the `wizard.stock.move.split`.
And here would be only  in `action_plan_reception` 
```
for move_to_split in self.move_to_split_ids
    self.move_ids |= move_to_split._split_move()
```

Reply:
But this is not a stopper. :-)